recipes/ruy/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+    homepage = "https://github.com/google/ruy"
+    license = "Apache-2.0"
+    topics = ("matrix", "multiplication", "neural", "network", "AI", "tensorflow")
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "3.4",
+            "apple-clang": "5.1",
+        }
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+
+        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn("Compiler is unknown. Assuming it supports C++14.")
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("Build requires support for C++14. Minimum version for {} is {}"
+                .format(str(self.settings.compiler), minimum_version))
+
+        if str(self.settings.compiler) == "clang" and tools.Version(self.settings.compiler.version) <= 5 and self.settings.build_type == "Debug":
+            raise ConanInvalidConfiguration("Debug builds are not supported on older versions of Clang (<=5)")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("cpuinfo/cci.20201217")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["RUY_MINIMAL_BUILD"] = True
+        self._cmake.configure()
+        return self._cmake
+
+
+    def build(self):
+        # 1. Allow Shared builds
+        tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "ruy_cc_library.cmake"),
+                              "add_library(${_NAME} STATIC",
+                              "add_library(${_NAME}"
+                              )
+        # 2. Shared builds fail with undefined symbols without this. clog is a library from the cpuinfo
+        tools.replace_in_file(os.path.join(self._source_subfolder, "ruy", "CMakeLists.txt"),
+                              "set(ruy_6_cpuinfo \"cpuinfo\")",
+                              "set(ruy_6_cpuinfo \"cpuinfo;clog\")"
+                              )
```

Comment:
It shouldn't break but I don't like hardcoded lib names of dependencies.

More robust solution I think, add this in CMakeLists wrapper:

```cmake
...
conan_basic_setup(TARGETS)

add_library(cpuinfo INTERFACE IMPORTED)
set_property(TARGET cpuinfo PROPERTY INTERFACE_LINK_LIBRARIES CONAN_PKG::cpuinfo)
```

Reply:
They're handling the libraries creation in a very weird way, otherwise I'd have done what you say.
I'll have to take a look the check if this would work.

Reply:
This did not work, but I removed the hardcoded libs anyways and replaced it with `self.deps_cpp_info["cpuinfo"].libs + self.deps_cpp_info["cpuinfo"].system_libs` which is more robust.

Reply:
https://github.com/conan-io/conan-center-index/pull/7187/commits/b2f5a2e0f04ee07e406fb0a3c746c0c1abf2ce2c